source/loader/layers/sanitizer/asan_libdevice.hpp

```diff
@@ -67,18 +67,20 @@ struct LocalArgsInfo {
     uint64_t SizeWithRedZone = 0;
 };
 
+constexpr std::size_t ASAN_MAX_NUM_REPORTS = 10;
+
 struct LaunchInfo {
-    uintptr_t PrivateShadowOffset =
-        0; // don't move this field, we use it in AddressSanitizerPass
+    // Don't move this field, we use it in AddressSanitizerPass
+    uintptr_t PrivateShadowOffset = 0;
 
     uintptr_t LocalShadowOffset = 0;
     uintptr_t LocalShadowOffsetEnd = 0;
-    DeviceSanitizerReport SanitizerReport;
 
     uint32_t NumLocalArgs = 0;
-    LocalArgsInfo *LocalArgs = nullptr; // ordered by ArgIndex
-};
+    LocalArgsInfo *LocalArgs = nullptr; // Ordered by ArgIndex
 
+    DeviceSanitizerReport SanitizerReport[ASAN_MAX_NUM_REPORTS];
```

Comment:
Where is this array used? I can't find any places where you index into this array.

Reply:
It's used in `libdevice/sanitizer_utils.cpp`, see LLVM pr.